chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -0,0 +1,564 @@
+            to_puzzle_hash=full_puzzle.get_tree_hash(),
+            amount=uint64(amount),
+            fee_amount=fee,
+            confirmed=False,
+            sent=uint32(0),
+            spend_bundle=full_spend,
+            additions=full_spend.additions(),
+            removals=full_spend.removals(),
+            wallet_id=self.wallet_info.id,
+            sent_to=[],
+            trade_id=None,
+            type=uint32(TransactionType.OUTGOING_TX.value),
+            name=bytes32(token_bytes()),
+            memos=list(compute_memos(full_spend).items()),
+        )
+        return nft_record
+
+    async def update_metadata(
+        self, nft_coin_info: NFTCoinInfo, uri: str, fee: uint64 = uint64(0)
+    ) -> Optional[SpendBundle]:
+        coin = nft_coin_info.coin
+        # we're not changing it
+
+        uncurried_nft = UncurriedNFT.uncurry(nft_coin_info.full_puzzle)
+
+        puzzle_hash = uncurried_nft.inner_puzzle.get_tree_hash()
+        condition_list = [make_create_coin_condition(puzzle_hash, coin.amount, [puzzle_hash])]
+        condition_list.append([int_to_bytes(-24), NFT_METADATA_UPDATER, uri.encode("utf-8")])
+
+        self.log.info("Attempting to add a url to NFT coin %s in the metadata: %s", nft_coin_info, uri)
+        inner_solution = solution_for_conditions(condition_list)
+        nft_tx_record = await self._make_nft_transaction(nft_coin_info, inner_solution, fee)
+        await self.standard_wallet.push_transaction(nft_tx_record)
+        return nft_tx_record.spend_bundle
+
+    async def transfer_nft(
+        self,
+        nft_coin_info: NFTCoinInfo,
+        puzzle_hash: bytes32,
+        did_hash=None,
+        fee: uint64 = uint64(0),
+    ) -> Optional[SpendBundle]:
+        self.log.debug("Attempt to transfer a new NFT")
+        coin = nft_coin_info.coin
+        self.log.debug("Transfering NFT coin %r to puzhash: %s", nft_coin_info, puzzle_hash)
+
+        amount = coin.amount
+        condition_list = [make_create_coin_condition(puzzle_hash, amount, [bytes32(puzzle_hash)])]
+        self.log.debug("Condition for new coin: %r", condition_list)
+        inner_solution = solution_for_conditions(condition_list)
+        nft_tx_record = await self._make_nft_transaction(nft_coin_info, inner_solution, fee)
+        await self.standard_wallet.push_transaction(nft_tx_record)
+        return nft_tx_record.spend_bundle
+
+    def get_current_nfts(self) -> List[NFTCoinInfo]:
+        return self.nft_wallet_info.my_nft_coins
+
+    async def save_info(self, nft_info: NFTWalletInfo, in_transaction: bool) -> None:
+        self.nft_wallet_info = nft_info
+        current_info = self.wallet_info
+        data_str = json.dumps(nft_info.to_json_dict())
+        wallet_info = WalletInfo(current_info.id, current_info.name, current_info.type, data_str)
+        self.wallet_info = wallet_info
+        await self.wallet_state_manager.user_store.update_wallet(wallet_info, in_transaction)
```

Comment:
Actually we should have the wallet versioned in general but since we don't have that at the moment and it will be a bigger change i think we should at least for now wrap `NFTWalletInfo` to give it a version since its stored in the DB and is likely to be changed? See https://github.com/Chia-Network/chia-blockchain/pull/11587 which could be used.